hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java

```diff
@@ -246,9 +241,15 @@ public ReconStorageContainerManagerFacade(OzoneConfiguration conf,
     IncrementalContainerReportHandler icrHandler =
         new ReconIncrementalContainerReportHandler(nodeManager,
             containerManager, scmContext);
+    long timeDuration = conf.getTimeDuration(
+        OzoneConfigKeys.OZONE_SCM_CLOSE_CONTAINER_WAIT_DURATION,
+        OzoneConfigKeys.OZONE_SCM_CLOSE_CONTAINER_WAIT_DURATION_DEFAULT
+            .getDuration(), TimeUnit.MILLISECONDS);
+    leaseManager = new LeaseManager<>("Lease Manager", timeDuration);
     CloseContainerEventHandler closeContainerHandler =
         new CloseContainerEventHandler(
-            pipelineManager, containerManager, scmContext);
+            pipelineManager, containerManager, scmContext,
+            leaseManager, timeDuration);
```

Comment:
Since Recon can't send close commands to DNs (`ReconNodeManager#onMessage`) this thread will be a no-op. Can we make a Recon specific version that does not run this thread? For example, Recon has its own handler for dead nodes: `ReconDeadNodeHandler`.

Reply:
@errose28 Its handled by defining leaseManager as nullable in constructor and passing it null for recon